### PR TITLE
Fix battle timescale

### PR DESCRIPTION
**Pull request by ThyWoof**
using a Postfix instead. Still taking into account battles only as well as multiplayer scenarios.

### CODE REVIEW

**Comment by ThyWoof** [comment]
need a campaign save file to fully test...

**Review by PhilPJL — COMMENTED** [review]
I assume this is ok, but not really sure how to test.

File: SolastaCommunityExpansion/Patches/GameUiBattle/GameTimeSetTimeScalePatcher.cs [hunk 1]
```diff
@@ -1,32 +1,24 @@
...
-            }
-            else
-            {
-                Time.timeScale = ___timeScale * (___fasterTimeMode ? Main.Settings.BattleCustomTimeScale : 1f);
-            }
-
-            return false;
+            ___timeScale = ___networkTimeScale != 1.0 
+                ? ___networkTimeScale 
+                : ___timeScale * (___fasterTimeMode || Main.Settings.PermanentlySpeedBattleUp ? Main.Settings.BattleCustomTimeScale : 1f);
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Patches/GameUiBattle/GameTimeSetTimeScalePatcher.cs:21]: If fasterTimeMode == true (where does this get set?) the timeScale is already x 1.5 and now we're speeding up more, is that correct?

Comment (ThyWoof) [reply SolastaCommunityExpansion/Patches/GameUiBattle/GameTimeSetTimeScalePatcher.cs:21]: Good catch. Should only consider the slider value

**Comment by ThyWoof** [comment]
> I assume this is ok, but not really sure how to test.
> If it didn't work previously how does this change things?

This is code that was written way before Sorcerer DLC. I took the latest implementation that now also considers MP. Making a postfix also simplifies the logic to only trigger in battles.

**Review by PhilPJL — APPROVED** [review]

**Pull request merged**